packages/messaging/src/controllers/sw-controller.ts

```diff
@@ -14,61 +14,67 @@
  * limitations under the License.
  */
 
+import './sw-types';
+
 import { FirebaseApp } from '@firebase/app-types';
 
 import { ERROR_CODES } from '../models/errors';
 import { DEFAULT_PUBLIC_VAPID_KEY } from '../models/fcm-details';
-import * as WorkerPageMessage from '../models/worker-page-message';
+import {
+  InternalMessage,
+  MessageParameter,
+  MessageType
+} from '../models/worker-page-message';
 import { ControllerInterface } from './controller-interface';
 
+// Let TS know that this is a service worker
+declare const self: ServiceWorkerGlobalScope;
+
 const FCM_MSG = 'FCM_MSG';
 
-export type BgMessageHandler = (input: any) => Promise<any>;
+export type BgMessageHandler = (input: Payload) => void;
```

Comment:
You could be expecting a promise to be returned here so you can pass the promise to event.waitUntil(). Otherwise the service worker could be terminated before the developers work has finished.

Reply:
That's a good point, thanks!